File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.java
class Solution {
    public int[] solution(String s) {
        int[] answer = new int[2];
        int count = 0;
        int times = 0;
        while (!s.equals("1")) {
            for (char c : s.toCharArray()) {
                if (c == '0')
                    count ++;
            }
            s = s.replace("0", "");
            int len = s.length();
            s = Integer.toBinaryString(len);
            times ++;
        }
        answer[0] = times;
        answer[1] = count;
        return answer;
    }
}

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.py
def solution(s):
    count: int = 0
    time: int = 0
    while s != '1':
        length: int = len(s)
        c: int = s.count('1')
        count += length - c
        s: str = '1' * count
        s: str = format(c, 'b')
        time += 1
    
    return [time, count]